src/DetailsView/components/should-show-report-export-button.ts

```diff
@@ -3,31 +3,25 @@
 
 import { VisualizationConfigurationFactory } from 'common/configs/visualization-configuration-factory';
 import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
-import { VisualizationStoreData } from 'common/types/store-data/visualization-store-data';
+import { TabStoreData } from 'common/types/store-data/tab-store-data';
 import { VisualizationType } from 'common/types/visualization-type';
 
 export interface ShouldShowReportExportButtonProps {
     visualizationConfigurationFactory: VisualizationConfigurationFactory;
     selectedTest: VisualizationType;
-    visualizationStoreData: VisualizationStoreData;
     featureFlagStoreData: FeatureFlagStoreData;
+    tabStoreData: TabStoreData;
 }
 
 export type ShouldShowReportExportButton = (props: ShouldShowReportExportButtonProps) => boolean;
 
-export function shouldShowReportExportButtonForAssessment(
-    props: ShouldShowReportExportButtonProps,
-): boolean {
+export function shouldShowReportExportButtonForAssessment(): boolean {
     return true;
 }
 
 export function shouldShowReportExportButtonForFastpass(
     props: ShouldShowReportExportButtonProps,
 ): boolean {
     const config = props.visualizationConfigurationFactory.getConfiguration(props.selectedTest);
-    const shouldShow = config.shouldShowExportReport(props.featureFlagStoreData);
-
-    const scanData = config.getStoreData(props.visualizationStoreData.tests);
-    const isEnabled = config.getTestStatus(scanData);
-    return shouldShow || isEnabled;
+    return config.shouldShowExportReport(props.tabStoreData, props.featureFlagStoreData);
```

Comment:
Can we have the `!tabStoreData.isChanged` check happen here instead of repeating it in each visualization config? The "does this config support exporting to reports?" part feels reasonable for the individual config to understand, but I don't think it feels right for each individual visualization to need to understand of how the target page URL state ties into the report export UX flow.

Reply:
This is a good call, I've updated to have the check happen here.